hack/ensure-opm.sh

```diff
@@ -5,18 +5,32 @@ set -e
 source hack/common.sh
 
 OPM_VERSION="v1.12.3"
-OPM_PLATFORM="linux-amd64-opm"
-if [ "$OS_TYPE" == "Darwin" ]; then
-	OPM_PLATFORM=darwin-amd64-opm
-fi
+OPM_ARCH=$(case $(uname -m) in x86_64) echo -n amd64 ;; aarch64) echo -n arm64 ;; *) echo -n "$(uname -m)" ;; esac)
+OPM_OS=$(uname | awk '{print tolower($0)}')
+OPM_PLATFORM="${OPM_OS}-${OPM_ARCH}-opm"
 OPM_URL="https://github.com/operator-framework/operator-registry/releases/download/${OPM_VERSION}/${OPM_PLATFORM}"
 OPM_BIN="opm-${OPM_VERSION}"
 OPM="${OUTDIR_TOOLS}/${OPM_BIN}"
-if [ ! -x "${OPM}" ]; then
-	echo "Downloading opm ${OPM_VERSION} for ${OS_TYPE}"
-	mkdir -p ${OUTDIR_TOOLS}
-	curl -JL ${OPM_URL} -o ${OPM}
-	chmod +x ${OPM}
-else
+OPERATOR_REGISTRY_TMPDIR="${OUTDIR_TOOLS}/operator-registry"
+
+if [ -x "${OPM}" ]; then
+	# Case 1: already has opm binray
 	echo "Using opm cached at ${OPM}"
-fi
\ No newline at end of file
+elif [ "${OPM_ARCH}" != "amd64" ]; then
+	# Case 2: image does not exist, need to build from source
```

Comment:
....wow, I can't believe this. :joy: But I see it is indeed necessary for now...